src/sagemaker/cli/compatibility/v2/modifiers/serde.py

```diff
@@ -0,0 +1,413 @@
+        return name in OLD_OBJECT_NAMES
+
+    def modify_node(self, node):
+        """Replaces the ``ast.Name`` node with a ``ast.Call`` node that
+        instantiates a class available in version 2.0 and later of the Python SDK:
+
+        - ``sagemaker.serializers.CSVSerializer()``
+        - ``sagemaker.serializers.JSONSerializer()``
+        - ``sagemaker.serializers.NumpySerializer()``
+        - ``sagemaker.deserializers.CSVDeserializer()``
+        - ``sagemaker.deserializers.JSONDeserializer()``
+        - ``sagemaker.deserializers.NumpyDeserializer()``
+
+        The ``sagemaker`` prefix is excluded from the returned node.
+
+        Args:
+            node (ast.Name): a node that represents a Python identifier.
+
+        Returns:
+            ast.Call: a node that represents the instantiation of a SerDe object.
+        """
+        object_name = node.id if isinstance(node, ast.Name) else node.attr
+        new_class_name = OLD_OBJECT_NAME_TO_NEW_CLASS_NAME[object_name]
+        namespace_name = NEW_CLASS_NAME_TO_NAMESPACES[new_class_name][0]
+        subpackage_name = namespace_name[namespace_name.find(".") + 1 :]
+        assert subpackage_name in {"serializers", "deserializers"}
+        return pasta.parse("%s.%s()" % (subpackage_name, new_class_name)).body[0].value
+
+
+class SerdeImportFromPredictorRenamer(Modifier):
+    """A class to update import statements starting with ``from sagemaker.predictor``."""
+
+    def node_should_be_modified(self, node):
+        """Checks if the import statement imports a SerDe from the
+        ``sagemaker.predictor`` module.
+
+        Args:
+            node (ast.ImportFrom): a node that represents a ``from ... import ... `` statement.
+                For more, see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            bool: True if and only if the ``ast.ImportFrom`` imports a SerDe
+                from the ``sagemaker.predictor`` module.
+        """
+        return node.module == "sagemaker.predictor" and any(
+            [name.name in (OLD_CLASS_NAMES | OLD_OBJECT_NAMES) for name in node.names]
+        )
+
+    def modify_node(self, node):
+        """Removes the imported SerDe classes, as applicable.
+
+        Args:
+            node (ast.ImportFrom): a node that represents a ``from ... import ... `` statement.
+                For more, see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            ast.ImportFrom: a node that represents a import statement, which has
+                been modified to remove imported serializers. If nothing is
+                imported, None is returned.
+        """
+        i = 0
+        while i < len(node.names):
+            name = node.names[i].name
+            if name in OLD_CLASS_NAMES | OLD_OBJECT_NAMES:
+                node.names.pop(i)
+            else:
+                i += 1
+
+        if not node.names:
+            return None
+
+        return node
```

Comment:
maybe positive statement instead?

```suggestion
        return node if node.names else None
```